webapp/src/ts/modules/contacts/contacts-edit.component.ts

```diff
@@ -184,46 +206,33 @@ export class ContactsEditComponent implements OnInit, OnDestroy, AfterViewInit {
       .then((result) => result.doc);
   }
 
-  private getForm(contact) {
+  private getForm(contact, contactType, titleKey) {
     let formId;
-    let titleKey;
-    const typeId = this.contactTypesService.getTypeId(contact) || this.routeSnapshot.params?.type;
-    return this.contactTypesService
-      .get(typeId)
-      .then(type => {
-        if (!type) {
-          console.error(`Unknown contact type "${typeId}"`);
-          return;
-        }
+    if (contact) { // editing
+      this.contact = contact;
+      this.contactId = contact._id;
+      formId = contactType.edit_form || contactType.create_form;
+    } else { // adding
+      this.contact = {
+        type: 'contact',
+        contact_type: this.routeSnapshot.params?.type,
+        parent: this.routeSnapshot.params?.parent_id || '',
+      };
+      this.contactId = null;
+      formId = contactType.create_form;
+    }
 
-        if (contact) { // editing
-          this.contact = contact;
-          this.contactId = contact._id;
-          titleKey = type.edit_key;
-          formId = type.edit_form || type.create_form;
-        } else { // adding
-          this.contact = {
-            type: 'contact',
-            contact_type: this.routeSnapshot.params?.type,
-            parent: this.routeSnapshot.params?.parent_id || '',
-          };
-          this.contactId = null;
-          formId = type.create_form;
-          titleKey = type.create_key;
+    this.translationsLoadedSubscription?.unsubscribe();
+    this.translationsLoadedSubscription = this.store
+      .select(Selectors.getTranslationsLoaded)
+      .subscribe((loaded) => {
+        if (loaded) {
+          this.translateService
+            .get(titleKey)
+            .then((title) => this.globalActions.setTitle(title));
         }
-
-        this.translationsLoadedSubscription?.unsubscribe();
-        this.translationsLoadedSubscription = this.store
-          .select(Selectors.getTranslationsLoaded)
-          .subscribe((loaded) => {
-            if (loaded) {
-              this.translateService
-                .get(titleKey)
-                .then((title) => this.globalActions.setTitle(title));
-            }
-          });
-        return formId;
       });
+    return formId;
   }
```

Comment:
Setting the title here is completely unrelated to what the rest of the function does. We can slim the arguments down to not include the title and set the title in another function. 
```suggestion
  private getForm(contact, contactType) {
    if (contact) { // editing
      this.contact = contact;
      this.contactId = contact._id;
      return contactType.edit_form || contactType.create_form;
    }

    // adding
    this.contact = {
      type: 'contact',
      contact_type: this.routeSnapshot.params?.type,
      parent: this.routeSnapshot.params?.parent_id || '',
    };
    this.contactId = null;
    return contactType.create_form;
  }

  private setTitle(titleKey) {
    this.translationsLoadedSubscription?.unsubscribe();
    this.translationsLoadedSubscription = this.store
      .select(Selectors.getTranslationsLoaded)
      .subscribe((loaded) => {
        if (loaded) {
          this.translateService
            .get(titleKey)
            .then((title) => this.globalActions.setTitle(title));
        }
      });
  }
```